### DISCUSSION TITLE
post/164

### DISCUSSION THREAD

**Discussion by giscus** [BY-juun/Blog#277] [Comments]
# post/164

NextJS v12의 Page Router 방식으로 개발된 개인블로그를 NextJS v13(14)의 AppRouter로 전환하면서 있었던 과정에 대해 정리한 포스트입니다.

https://byjuun.com/post/164

**Comment by geonwooPark** [comment]
안녕하세요. Next.js를 사용해서 저도 블로그를 만들고 있는데 궁금한 점이 있어서요! 혹시 SSG로 동작하는 페이지 내부 컴포넌트가 SSR로 동작하는 것은 불가능 한가요??

**Reply by june-by** [reply]
안녕하세요. <code>SSG</code>로 동작하는 페이지에서 <code>Dynamic</code>하게 동작하는 서버컴포넌트(SSR)를 사용하는 방법은 찾지 못했습니다.

SSG 페이지의 렌더링 방식을 SSG로 고정하기 위해 페이지 최상단에 <code>route-segment-config</code>의 <code>dynamic</code>을 적용해보는 방식도 사용해보았는데요. (https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamic)
빌드시, router path별로(generateStaticParams의 결과로 생성된) rsc payload 파일들이 생성되긴하지만 여전히 SSR방식으로 동작했습니다.
